app/assets/javascripts/components/MentionInput.jsx

```diff
@@ -0,0 +1,126 @@
+
+class MentionInput extends React.Component {
+
+    constructor(props) {
+      super();
+      this.state = {
+        users: [],
+        originalUsers: [],
+        idToUsers: {},
+        showUsers: false,
+        mentionedUsers: [],
+        value: ''
+      }
+    }
+
+    componentDidMount() {
+      Requester.get('/api/users/show').then((data) => {
+        this.setState({ users: data });
+        this.setState({ originalUsers: Array.from(data) })
+        let userIdToName = {};
+        data.forEach((user) => {
+            userIdToName[user.id] = user.first_name + " " + user.last_name;
+        });
+        this.setState({ idToUsers: userIdToName });
+      });
+    }
+    
+    userClicked = (evt) => {
+      const cursor = this.textInput.selectionStart;
+      const curr = this.textInput.value;
+      const closestAt = curr.lastIndexOf('@', cursor);
+      const currentTypedUser = curr.substring(closestAt + 1, cursor + 1);
+      const clickedUser= this.state.idToUsers[evt.target.id];
+      let value = '';
+      if (!!currentTypedUser) {
+        value = curr.substring(0, closestAt + 1) + clickedUser;
+      }
+      else {
+        value = this.state.value + clickedUser;
+      }
+      const newMentioned = this.state.mentionedUsers.concat(evt.target.id);
+      this.setState({
+        mentionedUsers: newMentioned,
+        showUsers: false,
+        value: value,
+        users: Array.from(this.state.originalUsers)
+      });
+    }
+
+    handleKeyPress = (event) => {
+      cursor = this.textInput.selectionStart;
+        if (event.charCode == 64) {
+          this.setState({
+            showUsers: true,
+            users: Array.from(this.state.originalUsers)
+          });
+        } 
```

Comment:
We should probably have the showUsers menu removed when someone types "@" then deletes it